synth-converter/src/graph/graph_builder.rs

```diff
@@ -0,0 +1,359 @@
+use crate::{
+    batch::{
+        Action, ActionName, Batch, Chemical, ContainerInfo, ContainerPosition, Observation, Sample,
+        SampleItem,
+    },
+    graph::{
+        namespaces::{alloqual, allores, cat, obo, purl, qudt, schema},
+        utils::generate_bnode_term,
+    },
+    rdf::rdf_serializers::{serialize_graph_to_jsonld, serialize_graph_to_turtle},
+};
+use anyhow::{Context, Result};
+use sophia::{
+    api::{
+        graph::MutableGraph,
+        ns::{rdf, xsd},
+    },
+    inmem::graph::LightGraph,
+};
+use sophia_api::{ns::NsTerm, term::SimpleTerm};
+
+/// An RDF Graph
+pub struct GraphBuilder {
+    graph: LightGraph,
+}
+
+/// Builds an RDF graph of Synthesis data for the cat+ ontology.
+///
+/// The rust structure "actions" in /parser/actions is mapped to the cat+ ontology
```

Comment:
```suggestion
/// The rust structure `actions` in /parser/actions is mapped to the cat+ ontology
```

suggestion: maybe we can link to `[Action]` somehow in the docs.
Also maybe generate the docs if possible, I have never done this. But might be worth a try.